backend/src/db/repositories/OAuth2Repository.ts

```diff
@@ -0,0 +1,393 @@
+import DB from '../DB';
+import {
+  OAuth2AuthorizationCodeRaw,
+  OAuth2ClientRaw,
+  OAuth2ConsentRaw,
+  OAuth2TokenRaw
+} from '../types/OAuth2';
+import {ResultSetHeader} from 'mysql2';
+import TokenService from '../../oauth/TokenService';
+import {config} from '../../config';
+import {AuthorizationCode, AuthorizationCodeModel, Falsey, Token} from 'oauth2-server';
+
+export default class OAuth2Repository implements AuthorizationCodeModel {
```

Comment:
just a thought, it would be cleaner to move `AuthorizationCodeModel` implementation into a dedicated class (it can have  `OAuth2Repository` or `OAuth2Manager` as a field) and keep `OAuth2Repository` responsibility limited to DB layer